sway-lsp/src/utils/keyword_docs.rs

```diff
@@ -0,0 +1,828 @@
+            /// When data follows along with a variant, such as with sway's built-in [`Option`] type, the data
+            /// is added as the type describes, for example `Option::Some(123)`. The same follows with
+            /// struct-like variants, with things looking like `ComplexEnum::LotsOfThings { usual_struct_stuff:
+            /// true, blah: 245, }`. Empty Enums are similar to [`!`] in that they cannot be
+            /// instantiated at all, and are used mainly to mess with the type system in interesting ways.
+            ///
+            /// [ADT]: https://en.wikipedia.org/wiki/Algebraic_data_type
+            mod enum_keyword {}
+        };
+
+        let self_keyword: ItemMod = parse_quote! {
+            /// The receiver of a method, or the current module.
+            ///
+            /// `self` is used in two situations: referencing the current module and marking
+            /// the receiver of a method.
+            ///
+            /// In paths, `self` can be used to refer to the current module, either in a
+            /// [`use`] statement or in a path to access an element:
+            ///
+            /// ```sway
+            /// use std::contract_id::{self, ContractId};
+            /// ```
+            ///
+            /// Is functionally the same as:
+            ///
+            /// ```sway
+            /// use std::contract_id;
+            /// use std::contract_id::ContractId;
+            /// ```
+            ///
+            /// `self` as the current receiver for a method allows to omit the parameter
+            /// type most of the time. With the exception of this particularity, `self` is
+            /// used much like any other parameter:
+            ///
+            /// ```sway
+            /// struct Foo(u32);
+            ///
+            /// impl Foo {
+            ///     // No `self`.
+            ///     fn new() -> Self {
+            ///         Self(0)
+            ///     }
+            ///
+            ///     // Borrowing `self`.
+            ///     fn value(&self) -> u32 {
+            ///         self.0
+            ///     }
+            ///
+            ///     // Updating `self` mutably.
+            ///     fn clear(ref mut self) {
+            ///         self.0 = 0
+            ///     }
+            /// }
+            /// ```
+            mod self_keyword {}
+        };
+
+        let fn_keyword: ItemMod = parse_quote! {
+            /// Functions are the primary way code is executed within Sway. Function blocks, usually just
+            /// called functions, can be defined in a variety of different places and be assigned many
+            /// different attributes and modifiers.
+            ///
+            /// Standalone functions that just sit within a module not attached to anything else are common,
+            /// but most functions will end up being inside [`impl`] blocks, either on another type itself, or
+            /// as a trait impl for that type.
```

Comment:
The formatting gets messed up on this line apparently because the line is 1 character too long. Moving "or" to the next line fixes it.


<img width="322" alt="image" src="https://user-images.githubusercontent.com/47993817/215905612-30ded812-998e-46d3-9f54-e0a26e0f9f07.png">

```suggestion
            /// but most functions will end up being inside [`impl`] blocks, either on another type itself,
            /// or as a trait impl for that type.
```

<img width="319" alt="image" src="https://user-images.githubusercontent.com/47993817/215905954-888888fb-30de-4f78-ad56-7e31f75412fc.png">

Is there a way to detect this with clippy/rustfmt? It could be a problem in other places.

Reply:
thanks for finding this, turns out there was a bug in the `extract_lit` function. I've updated the logic so this can't happen anymore. :)